### ISSUE TITLE
Feature request: Ability to skip lines that start with a certain character

### ISSUE THREAD

**Issue by WardBrian** [YS-L/csvlens#52]
Hi!

I regularly deal with CSV files that have "comments" in them, lines which (in my case) start with `#` and do not contain either a header or a comma separated row. It seems that at present this can result in csvlens believing there is only one column in the file.

Would it be reasonable to support this with an extra command line flag?

**Comment by hbina** [comment]
Can't you just grep them out via

```
grep '#' <file> -v | csvlens
```

**Comment by ggrothendieck** [comment]
Another possibility is to have a command line flag that allows specifying a string such that all lines before the first line encountered with that string are skipped over.  The string can be specified as any string in the header that is unlikely to be in the lines before it.

In the R language, the data.table::fread function works like that.

csv files from Excel often have junk at the top and there is no distinguishing starting character and this would handle that case too.

Of course `gawk '/mystring/,0' myfile.csv | csvlens`  or `sed -n '/mystring/,$p' myfile.csv | csvlens` work but the mental load is higher than `csvlens -s mystring myfile.csv`

```

```

**Comment by Knusper** [comment]
> Can't you just grep them out via
> 
> ```
> grep '#' <file> -v | csvlens
> ```

Yes this works.  But then in the bottom right corner it will display `STDIN` and not the filename, which can be confusing.
